include/nanobind/eigen/sparse.h

```diff
@@ -0,0 +1,115 @@
+/*
+    nanobind/eigen/dense.h: type casters for dense Eigen
+    vectors and matrices
+
+    Copyright (c) 2023 Wenzel Jakob
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <nanobind/tensor.h>
+#include <nanobind/eigen/dense.h>
+#include <Eigen/SparseCore>
+
+static_assert(EIGEN_VERSION_AT_LEAST(3, 3, 0),
```

Comment:
I don't think you need to repeat the version check, if it is already in `eigen/dense.h`.